tests/rule_based_profiler/test_rule_based_profiler.py

```diff
@@ -1211,3 +1272,48 @@ def test_list_profilers_in_cloud_mode(mock_profiler_store: mock.MagicMock):
 
     assert res == keys
     assert store.list_keys.called
+
+
+@mock.patch("great_expectations.data_context.data_context.DataContext")
+@mock.patch(
+    "great_expectations.rule_based_profiler.domain_builder.SimpleColumnSuffixDomainBuilder"
+)
+@mock.patch(
+    "great_expectations.rule_based_profiler.expectation_configuration_builder.DefaultExpectationConfigurationBuilder"
+)
+def test_add_rule(
+    mock_data_context: mock.MagicMock,
+    mock_domain_builder: mock.MagicMock,
+    mock_expectation_configuration_builder: mock.MagicMock,
+    sample_rule_dict: dict,
+):
+
+    profiler: RuleBasedProfiler = RuleBasedProfiler(
+        name="my_mocked_rbp", data_context=mock_data_context, config_version=1.0
+    )
+    first_rule: Rule = Rule(
+        name="first_rule",
+        domain_builder=mock_domain_builder,
+        expectation_configuration_builders=mock_expectation_configuration_builder,
+    )
+    first_rule.to_json_dict = MagicMock(return_value=sample_rule_dict)
+    profiler.add_rule(first_rule)
+    assert len(profiler.rules) == 1
+
+    duplicate_of_first_rule: Rule = Rule(
+        name="first_rule",
+        domain_builder=mock_domain_builder,
+        expectation_configuration_builders=mock_expectation_configuration_builder,
+    )
+    duplicate_of_first_rule.to_json_dict = MagicMock(return_value=sample_rule_dict)
+    profiler.add_rule(duplicate_of_first_rule)
+    assert len(profiler.rules) == 1
+
+    second_rule: Rule = Rule(
+        name="second_rule",
+        domain_builder=mock_domain_builder,
+        expectation_configuration_builders=mock_expectation_configuration_builder,
+    )
+    second_rule.to_json_dict = MagicMock(return_value=sample_rule_dict)
+    profiler.add_rule(second_rule)
+    assert len(profiler.rules) == 2
```

Comment:
I think we should separate this out into 2 tests - one for the case where we remove a rule with the same name and one where we simply append.

As a general rule of thumb, I try to write a unit test for every possible code path. So with something like this:
```python
if x:
    ...
elif y:
    ...
else:
    raise Exception(...)
```
I'd write 3 separate tests!